### ISSUE TITLE
Tests failing on master branch

### ISSUE THREAD

**Issue by aryadas98** [sblunt/orbitize#152]
Tests on the latest master branch are failing. From the root directory of the project if I run `pytest` these errors come up:

```
======================================= test session starts ========================================
platform linux -- Python 3.6.7, pytest-5.3.4, py-1.8.1, pluggy-0.13.1
rootdir: /home/arya/orbitize
collected 18 items / 6 errors / 12 selected

============================================== ERRORS ==============================================
_______________________________ ERROR collecting tests/test_OFTI.py ________________________________
venv/lib/python3.6/site-packages/numpy/core/function_base.py:117: in linspace
    num = operator.index(num)
E   TypeError: 'float' object cannot be interpreted as an integer

During handling of the above exception, another exception occurred:
tests/test_OFTI.py:10: in <module>
    import orbitize.sampler as sampler
orbitize/sampler.py:19: in <module>
    import orbitize.results
orbitize/results.py:27: in <module>
    cmap(np.linspace(0.0, 0.7, 1000.))
<__array_function__ internals>:6: in linspace
    ???
venv/lib/python3.6/site-packages/numpy/core/function_base.py:121: in linspace
    .format(type(num)))
E   TypeError: object of type <class 'float'> cannot be safely interpreted as an integer.
```

The rest of the errors are caused by the same line `cmap(np.linspace(0.0, 0.7, 1000.))`. The problem is that the last argument 1000 should be an integer. However even after removing the period, the tests are failing.

After removing period:
```
======================================= test session starts ========================================
platform linux -- Python 3.6.7, pytest-5.3.4, py-1.8.1, pluggy-0.13.1
rootdir: /home/arya/orbitize
collected 38 items

tests/test_OFTI.py ...                                                                       [  7%]
tests/test_api.py FF...                                                                      [ 21%]
tests/test_conversions.py ...                                                                [ 28%]
tests/test_driver.py FF.                                                                     [ 36%]
tests/test_kepler_solver.py .........                                                        [ 60%]
tests/test_mcmc.py F.                                                                        [ 65%]
tests/test_priors.py ..                                                                      [ 71%]
tests/test_read_input.py FFFF                                                                [ 81%]
tests/test_results.py ....                                                                   [ 92%]
tests/test_system.py ...                                                                     [100%]

============================================= FAILURES =============================================
________________________________________ test_compute_model ________________________________________

    def test_compute_model():
        """
        Test basic functionality of ``System.compute_model()``
        """
        testdir = os.path.dirname(os.path.abspath(__file__))
        input_file = os.path.join(testdir, 'test_val.csv')
        data_table = read_input.read_file(input_file)
        data_table['object'] = 1
        testSystem_parsing = system.System(
            1, data_table, 10., 10.
        )

        params_arr = np.array([[1.,0.5],[0.,0.],[0.,0.],[0.,0.],[0.,0.],[245000., 245000.], [10, 10], [10, 10]])
        model = testSystem_parsing.compute_model(params_arr)
>       assert model.shape == (4,2,2)
E       assert (8, 2, 2) == (4, 2, 2)
E         At index 0 diff: 8 != 4
E         Use -v to get the full diff

tests/test_api.py:25: AssertionError
_________________________________________ test_systeminit __________________________________________

    def test_systeminit():
        """
        Test that initializing a ``System`` class produces a list of ``Prior``
        objects of the correct length when:
            - parallax and total mass are fixed
            - parallax and total mass errors are given
            - parallax is fixed, total mass error is given
            - parallax error is given, total mass error is fixed

        Test that the different types of data are parsed correctly
        when initializing a ``System`` object.
        """
        testdir = os.path.dirname(os.path.abspath(__file__))
        input_file = os.path.join(testdir, 'test_val.csv')
        data_table = read_input.read_file(input_file)

        # Manually set 'object' column of data table
        data_table['object'] = 1
        data_table['object'][1] = 2

        plx_mass_errs2lens = {
            (0.,0.): 14,
            (1.,1.): 14,
            (0.,1.): 14,
            (1.,0.): 14
        }

        for plx_e, mass_e in plx_mass_errs2lens.keys():

            testSystem_priors = system.System(
                2, data_table, 10., 10., plx_err=plx_e, mass_err=mass_e
            )
            assert len(testSystem_priors.sys_priors) == \
                plx_mass_errs2lens[(plx_e, mass_e)]

        testSystem_parsing = system.System(
            2, data_table, 10., 10.,
            plx_err=0.5, mass_err=0.5
        )
        assert len(data_table[testSystem_parsing.seppa[0]]) == 0
>       assert len(data_table[testSystem_parsing.seppa[1]]) == 1
E       assert 0 == 1
E        +  where 0 = len(<Table length=0>\n epoch  object  quant1 quant1_err  quant2 quant2_err quant_type\nfloat64 int64  float64  float64   float64  float64     bytes5  \n------- ------ ------- ---------- ------- ---------- ----------)

tests/test_api.py:71: AssertionError
_________________________________ test_create_driver_from_filename _________________________________

    def test_create_driver_from_filename():
        """
        Test creation of Driver object from filename as input
        """
        testdir = os.path.dirname(os.path.abspath(__file__))
        input_file = os.path.join(testdir, 'test_val.csv')
        myDriver = driver.Driver(input_file, # path to data file
                                 'MCMC', # name of algorith for orbit-fitting
                                 1, # number of secondary bodies in system
                                 1.0, # total system mass [M_sun]
                                 50.0, # total parallax of system [mas]
                                 mass_err=0.1, # mass error [M_sun]
                                 plx_err=0.1) # parallax error [mas]
>       _compare_table(myDriver.system.data_table)

tests/test_driver.py:71:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

input_table = <Table length=8>
 epoch  object  quant1 quant1_err  quant2 quant2_err quant_type
float64 int64  float64  float64   flo...     0     0.0        0.0     0.0        0.0      radec
 1237.0      0    10.0        0.1     nan        nan         rv

    def _compare_table(input_table):
        """
        Tests input table to expected values, which are:
            epoch  object  quant1 quant1_err  quant2 quant2_err quant_type
           float64  int   float64  float64   float64  float64      str5
           ------- ------ ------- ---------- ------- ---------- ----------
           1234.0      1    0.01      0.005     0.5       0.05      radec
           1235.0      1     1.0      0.005    89.0        0.1      seppa
           1236.0      1     1.0      0.005    89.3        0.3      seppa
           1237.0      0    10.0        0.1     nan        nan         rv
        """
        rows_expected = 4
        epoch_expected = [1234, 1235, 1236, 1237]
        object_expected = [1,1,1,0]
        quant1_expected = [0.01, 1.0, 1.0, 10.0]
        quant1_err_expected = [0.005, 0.005, 0.005, 0.1]
        quant2_expected = [0.5, 89.0, 89.3, np.nan]
        quant2_err_expected = [0.05, 0.1, 0.3, np.nan]
        quant_type_expected = ['radec', 'seppa', 'seppa', 'rv']
>       assert len(input_table) == rows_expected
E       assert 8 == 4
E        +  where 8 = len(<Table length=8>\n epoch  object  quant1 quant1_err  quant2 quant2_err quant_type\nfloat64 int64  float64  float64   flo...     0     0.0        0.0     0.0        0.0      radec\n 1237.0      0    10.0        0.1     nan        nan         rv)

tests/test_driver.py:30: AssertionError
__________________________________ test_create_driver_from_table ___________________________________

    def test_create_driver_from_table():
        """
        Test creation of Driver object from Table as input
        """
        testdir = os.path.dirname(os.path.abspath(__file__))
        input_file = os.path.join(testdir, 'test_val.csv')
        input_table = read_file(input_file)
        myDriver = driver.Driver(input_table, # astropy.table Table of input
                                 'MCMC', # name of algorithm for orbit-fitting
                                 1, # number of secondary bodies in system
                                 1.0, # total system mass [M_sun]
                                 50.0, # total parallax of system [mas]
                                 mass_err=0.1, # mass error [M_sun]
                                 plx_err=0.1) # parallax error [mas]
>       _compare_table(myDriver.system.data_table)

tests/test_driver.py:88:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

input_table = <Table length=8>
 epoch  object  quant1 quant1_err  quant2 quant2_err quant_type
float64 int64  float64  float64   flo...     0     0.0        0.0     0.0        0.0      radec
 1237.0      0    10.0        0.1     nan        nan         rv

    def _compare_table(input_table):
        """
        Tests input table to expected values, which are:
            epoch  object  quant1 quant1_err  quant2 quant2_err quant_type
           float64  int   float64  float64   float64  float64      str5
           ------- ------ ------- ---------- ------- ---------- ----------
           1234.0      1    0.01      0.005     0.5       0.05      radec
           1235.0      1     1.0      0.005    89.0        0.1      seppa
           1236.0      1     1.0      0.005    89.3        0.3      seppa
           1237.0      0    10.0        0.1     nan        nan         rv
        """
        rows_expected = 4
        epoch_expected = [1234, 1235, 1236, 1237]
        object_expected = [1,1,1,0]
        quant1_expected = [0.01, 1.0, 1.0, 10.0]
        quant1_err_expected = [0.005, 0.005, 0.005, 0.1]
        quant2_expected = [0.5, 89.0, 89.3, np.nan]
        quant2_err_expected = [0.05, 0.1, 0.3, np.nan]
        quant_type_expected = ['radec', 'seppa', 'seppa', 'rv']
>       assert len(input_table) == rows_expected
E       assert 8 == 4
E        +  where 8 = len(<Table length=8>\n epoch  object  quant1 quant1_err  quant2 quant2_err quant_type\nfloat64 int64  float64  float64   flo...     0     0.0        0.0     0.0        0.0      radec\n 1237.0      0    10.0        0.1     nan        nan         rv)

tests/test_driver.py:30: AssertionError
__________________________________________ test_mcmc_runs __________________________________________

num_temps = 0, num_threads = 1

    def test_mcmc_runs(num_temps=0, num_threads=1):
        """
        Tests the MCMC sampler by making sure it even runs
        Args:
            num_temps: Number of temperatures to use
                Uses Parallel Tempering MCMC (ptemcee) if > 1,
                otherwises, uses Affine-Invariant Ensemble Sampler (emcee)
            num_threads: number of threads to run
        """

        # use the test_csv dir
        testdir = os.path.dirname(os.path.abspath(__file__))
        input_file = os.path.join(testdir, 'test_val.csv')

        # construct Driver
        n_walkers=100
        myDriver = Driver(input_file, 'MCMC', 1, 1, 0.01,
            mcmc_kwargs={'num_temps':2, 'num_threads':num_threads, 'num_walkers':n_walkers}
        )

        # run it a little (tests 0 burn-in steps)
>       myDriver.sampler.run_sampler(100)

tests/test_mcmc.py:31:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
orbitize/sampler.py:592: in run_sampler
    for pos, lnprob, lnlike in sampler.sample(self.curr_pos, iterations=burn_steps, thin=thin):
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <ptemcee.sampler.Sampler object at 0x7f8f2616ea90>
p0 = array([[[3.29810431e+04, 6.76064840e-01, 9.93403356e-01, 1.99182241e+00,
         5.29050755e+00, 4.43971309e-01],
   ...],
        [2.58972568e-01, 1.06527851e-01, 2.19137938e+00, 1.27295714e-01,
         1.56220450e+00, 6.76397888e-01]]])
iterations = 0, thin = 1, storechain = True, adapt = False, swap_ratios = False

    def sample(self, p0=None,
               iterations=1, thin=1,
               storechain=True, adapt=False,
               swap_ratios=False):
        """
        Advance the chains ``iterations`` steps as a generator.

        :param p0:
            The initial positions of the walkers.  Shape should be
            ``(ntemps, nwalkers, dim)``.  Can be omitted if resuming
            from a previous run.

        :param iterations: (optional)
            The number of iterations to perform.

        :param thin: (optional)
            The number of iterations to perform between saving the
            state to the internal chain.

        :param storechain: (optional)
            If ``True`` store the iterations in the ``chain``
            property.

        :param adapt: (optional)
            If ``True``, the temperature ladder is dynamically adapted as the sampler runs to
            achieve uniform swap acceptance ratios between adjacent chains.  See `arXiv:1501.05823
            <http://arxiv.org/abs/1501.05823>`_ for details.

        :param swap_ratios: (optional)
            If ``True``, also yield the instantaneous inter-chain acceptance ratios in the fourth
            element of the yielded tuple.

        At each iteration, this generator yields

        * ``p``, the current position of the walkers.

        * ``logpost``, the current posterior values for the walkers.

        * ``loglike``, the current likelihood values for the walkers.

        * ``ratios``, the instantaneous inter-chain acceptance ratios (if requested).

        """

        # Set initial walker positions.
        if p0 is not None:
            # Start anew.
            self._p0 = p = np.array(p0).copy()
            self._logposterior0 = None
            self._loglikelihood0 = None
        elif self._p0 is not None:
            # Now, where were we?
            p = self._p0
        else:
            raise ValueError('Initial walker positions not specified.')

        # Check for dodgy inputs.
        if np.any(np.isinf(p)):
            raise ValueError('At least one parameter value was infinite.')
        if np.any(np.isnan(p)):
            raise ValueError('At least one parameter value was NaN.')

        # If we have no likelihood or prior values, compute them.
        if self._logposterior0 is None or self._loglikelihood0 is None:
            logl, logp = self._evaluate(p)
            logpost = self._tempered_likelihood(logl) + logp

            self._loglikelihood0 = logl
            self._logposterior0 = logpost
        else:
            logl = self._loglikelihood0
            logpost = self._logposterior0

        if (logpost == -np.inf).any():
>           raise ValueError('Attempting to start with samples outside posterior support.')
E           ValueError: Attempting to start with samples outside posterior support.

venv/lib/python3.6/site-packages/ptemcee/sampler.py:355: ValueError
__________________________________________ test_read_file __________________________________________

    def test_read_file():
        """
        Test the read_file function using the test_val.csv file and test_val_radec.csv
        """
        testdir = os.path.dirname(os.path.abspath(__file__))
        # Check that main test input is read in with correct values
        input_file = os.path.join(testdir, 'test_val.csv')
>       _compare_table(read_file(input_file))

tests/test_read_input.py:62:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

input_table = <Table length=8>
 epoch  object  quant1 quant1_err  quant2 quant2_err quant_type
float64 int64  float64  float64   flo...     0     0.0        0.0     0.0        0.0      radec
 1237.0      0    10.0        0.1     nan        nan         rv

    def _compare_table(input_table):
        """
        Tests input table to expected values, which are:
            epoch  object  quant1 quant1_err  quant2 quant2_err quant_type
           float64  int   float64  float64   float64  float64      str5
           ------- ------ ------- ---------- ------- ---------- ----------
           1234.0      1    0.01      0.005     0.5       0.05      radec
           1235.0      1     1.0      0.005    89.0        0.1      seppa
           1236.0      1     1.0      0.005    89.3        0.3      seppa
           1237.0      0    10.0        0.1     nan        nan         rv
        """
        rows_expected = 4
        epoch_expected = [1234, 1235, 1236, 1237]
        object_expected = [1,1,1,0]
        quant1_expected = [0.01, 1.0, 1.0, 10.0]
        quant1_err_expected = [0.005, 0.005, 0.005, 0.1]
        quant2_expected = [0.5, 89.0, 89.3, np.nan]
        quant2_err_expected = [0.05, 0.1, 0.3, np.nan]
        quant_type_expected = ['radec', 'seppa', 'seppa', 'rv']
>       assert len(input_table) == rows_expected
E       assert 8 == 4
E        +  where 8 = len(<Table length=8>\n epoch  object  quant1 quant1_err  quant2 quant2_err quant_type\nfloat64 int64  float64  float64   flo...     0     0.0        0.0     0.0        0.0      radec\n 1237.0      0    10.0        0.1     nan        nan         rv)

tests/test_read_input.py:27: AssertionError
____________________________________ test_write_orbitize_input _____________________________________

    def test_write_orbitize_input():
        """
        Test the write_orbitize_input and the read_file functions
        """
        testdir = os.path.dirname(os.path.abspath(__file__))
        input_file = os.path.join(testdir, 'test_val.csv')
        test_table = read_file(input_file)
        output_file = os.path.join(testdir, 'temp_test_orbitize_input.csv')
        # If temp output file already exists, delete it
        if os.path.isfile(output_file):
            os.remove(output_file)
        try:  # Catch these tests so that we remove temporary file
            # Test that we were able to write the table
            write_orbitize_input(test_table,output_file)
            assert os.path.isfile(output_file)
            # Test that we can read the table and check if it's correct
            test_table_2 = read_file(output_file)
>           _compare_table(test_table_2)

tests/test_read_input.py:99:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

input_table = <Table length=8>
 epoch  object  quant1 quant1_err  quant2 quant2_err quant_type
float64 int64  float64  float64   flo...     0     0.0        0.0     0.0        0.0      radec
 1237.0      0    10.0        0.1     nan        nan         rv

    def _compare_table(input_table):
        """
        Tests input table to expected values, which are:
            epoch  object  quant1 quant1_err  quant2 quant2_err quant_type
           float64  int   float64  float64   float64  float64      str5
           ------- ------ ------- ---------- ------- ---------- ----------
           1234.0      1    0.01      0.005     0.5       0.05      radec
           1235.0      1     1.0      0.005    89.0        0.1      seppa
           1236.0      1     1.0      0.005    89.3        0.3      seppa
           1237.0      0    10.0        0.1     nan        nan         rv
        """
        rows_expected = 4
        epoch_expected = [1234, 1235, 1236, 1237]
        object_expected = [1,1,1,0]
        quant1_expected = [0.01, 1.0, 1.0, 10.0]
        quant1_err_expected = [0.005, 0.005, 0.005, 0.1]
        quant2_expected = [0.5, 89.0, 89.3, np.nan]
        quant2_err_expected = [0.05, 0.1, 0.3, np.nan]
        quant_type_expected = ['radec', 'seppa', 'seppa', 'rv']
>       assert len(input_table) == rows_expected
E       assert 8 == 4
E        +  where 8 = len(<Table length=8>\n epoch  object  quant1 quant1_err  quant2 quant2_err quant_type\nfloat64 int64  float64  float64   flo...     0     0.0        0.0     0.0        0.0      radec\n 1237.0      0    10.0        0.1     nan        nan         rv)

tests/test_read_input.py:27: AssertionError
_____________________________________ test_read_formatted_file _____________________________________

args = (), kwargs = {}, caught_warnings = [<warnings.WarningMessage object at 0x7f8f470742e8>]

    def test_inner(*args, **kwargs):
        with warnings.catch_warnings(record=True) as caught_warnings:
            warnings.simplefilter("always")
>           rv = method(*args, **kwargs)

venv/lib/python3.6/site-packages/deprecation.py:268:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
tests/test_read_input.py:77: in test_read_formatted_file
    _compare_table(read_formatted_file(input_file))
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

input_table = <Table length=8>
 epoch  object  quant1 quant1_err  quant2 quant2_err quant_type
float64 int64  float64  float64   flo...     0     0.0        0.0     0.0        0.0      radec
 1237.0      0    10.0        0.1     nan        nan         rv

    def _compare_table(input_table):
        """
        Tests input table to expected values, which are:
            epoch  object  quant1 quant1_err  quant2 quant2_err quant_type
           float64  int   float64  float64   float64  float64      str5
           ------- ------ ------- ---------- ------- ---------- ----------
           1234.0      1    0.01      0.005     0.5       0.05      radec
           1235.0      1     1.0      0.005    89.0        0.1      seppa
           1236.0      1     1.0      0.005    89.3        0.3      seppa
           1237.0      0    10.0        0.1     nan        nan         rv
        """
        rows_expected = 4
        epoch_expected = [1234, 1235, 1236, 1237]
        object_expected = [1,1,1,0]
        quant1_expected = [0.01, 1.0, 1.0, 10.0]
        quant1_err_expected = [0.005, 0.005, 0.005, 0.1]
        quant2_expected = [0.5, 89.0, 89.3, np.nan]
        quant2_err_expected = [0.05, 0.1, 0.3, np.nan]
        quant_type_expected = ['radec', 'seppa', 'seppa', 'rv']
>       assert len(input_table) == rows_expected
E       assert 8 == 4
E        +  where 8 = len(<Table length=8>\n epoch  object  quant1 quant1_err  quant2 quant2_err quant_type\nfloat64 int64  float64  float64   flo...     0     0.0        0.0     0.0        0.0      radec\n 1237.0      0    10.0        0.1     nan        nan         rv)

tests/test_read_input.py:27: AssertionError
___________________________________ test_write_orbitize_input_2 ____________________________________

args = (), kwargs = {}, caught_warnings = [<warnings.WarningMessage object at 0x7f8f263d89e8>]

    def test_inner(*args, **kwargs):
        with warnings.catch_warnings(record=True) as caught_warnings:
            warnings.simplefilter("always")
>           rv = method(*args, **kwargs)

venv/lib/python3.6/site-packages/deprecation.py:268:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
tests/test_read_input.py:124: in test_write_orbitize_input_2
    _compare_table(test_table_2)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

input_table = <Table length=8>
 epoch  object  quant1 quant1_err  quant2 quant2_err quant_type
float64 int64  float64  float64   flo...     0     0.0        0.0     0.0        0.0      radec
 1237.0      0    10.0        0.1     nan        nan         rv

    def _compare_table(input_table):
        """
        Tests input table to expected values, which are:
            epoch  object  quant1 quant1_err  quant2 quant2_err quant_type
           float64  int   float64  float64   float64  float64      str5
           ------- ------ ------- ---------- ------- ---------- ----------
           1234.0      1    0.01      0.005     0.5       0.05      radec
           1235.0      1     1.0      0.005    89.0        0.1      seppa
           1236.0      1     1.0      0.005    89.3        0.3      seppa
           1237.0      0    10.0        0.1     nan        nan         rv
        """
        rows_expected = 4
        epoch_expected = [1234, 1235, 1236, 1237]
        object_expected = [1,1,1,0]
        quant1_expected = [0.01, 1.0, 1.0, 10.0]
        quant1_err_expected = [0.005, 0.005, 0.005, 0.1]
        quant2_expected = [0.5, 89.0, 89.3, np.nan]
        quant2_err_expected = [0.05, 0.1, 0.3, np.nan]
        quant_type_expected = ['radec', 'seppa', 'seppa', 'rv']
>       assert len(input_table) == rows_expected
E       assert 8 == 4
E        +  where 8 = len(<Table length=8>\n epoch  object  quant1 quant1_err  quant2 quant2_err quant_type\nfloat64 int64  float64  float64   flo...     0     0.0        0.0     0.0        0.0      radec\n 1237.0      0    10.0        0.1     nan        nan         rv)

tests/test_read_input.py:27: AssertionError
========================================= warnings summary =========================================
orbitize/results.py:223
  /home/arya/orbitize/orbitize/results.py:223: DeprecationWarning: invalid escape sequence \c
    'inc':'inc [$^{\circ}$]',

orbitize/results.py:224
  /home/arya/orbitize/orbitize/results.py:224: DeprecationWarning: invalid escape sequence \o
    'aop':'$\omega$ [$^{\circ}$]',

orbitize/results.py:225
  /home/arya/orbitize/orbitize/results.py:225: DeprecationWarning: invalid escape sequence \O
    'pan':'$\Omega$ [$^{\circ}$]',

orbitize/results.py:227
  /home/arya/orbitize/orbitize/results.py:227: DeprecationWarning: invalid escape sequence \p
    'plx':'$\pi$ [mas]',

orbitize/results.py:228
  /home/arya/orbitize/orbitize/results.py:228: DeprecationWarning: invalid escape sequence \o
    'mtot':'$M_T$ [M$_{\odot}$]',

orbitize/results.py:229
  /home/arya/orbitize/orbitize/results.py:229: DeprecationWarning: invalid escape sequence \o
    'm0':'$M_0$ [M$_{\odot}$]',

orbitize/results.py:230
  /home/arya/orbitize/orbitize/results.py:230: DeprecationWarning: invalid escape sequence \o
    'm1':'$M_1$ [M$_{\odot}$]',

orbitize/results.py:418
  /home/arya/orbitize/orbitize/results.py:418: DeprecationWarning: invalid escape sequence \D
    ax.set_xlabel('$\Delta$RA [mas]')

orbitize/results.py:419
  /home/arya/orbitize/orbitize/results.py:419: DeprecationWarning: invalid escape sequence \D
    ax.set_ylabel('$\Delta$Dec [mas]')

tests/test_api.py::test_custom_likelihood
  /home/arya/orbitize/tests/test_api.py:117: DeprecatedWarning: read_formatted_file is deprecated as of 1.0.2 and will be removed in 2.0. Use read_file() instead. v1.0.2 replaces read_formatted_file and read_orbitize_input with read_file(). For now, this will be a wrapper for read_file and will be removed in the v2.0 release.
    data_table = read_input.read_formatted_file(input_file)

tests/test_mcmc.py::test_mcmc_runs
tests/test_mcmc.py::test_examine_chop_chains
  /home/arya/orbitize/orbitize/lnlike.py:46: RuntimeWarning: divide by zero encountered in true_divide
    chi2 = -0.5 * residual**2 / errors**2

tests/test_mcmc.py::test_mcmc_runs
tests/test_mcmc.py::test_examine_chop_chains
  /home/arya/orbitize/orbitize/lnlike.py:46: RuntimeWarning: invalid value encountered in true_divide
    chi2 = -0.5 * residual**2 / errors**2

tests/test_mcmc.py::test_examine_chop_chains
tests/test_priors.py::test_compute_lnprob
  /home/arya/orbitize/orbitize/priors.py:163: RuntimeWarning: invalid value encountered in log
    lnprob = -np.log((element_array*normalizer))

tests/test_mcmc.py::test_examine_chop_chains
  /home/arya/orbitize/venv/lib/python3.6/site-packages/emcee/ensemble.py:335: RuntimeWarning: invalid value encountered in subtract
    lnpdiff = (self.dim - 1.) * np.log(zz) + newlnprob - lnprob0

tests/test_mcmc.py::test_examine_chop_chains
  /home/arya/orbitize/venv/lib/python3.6/site-packages/emcee/ensemble.py:336: RuntimeWarning: invalid value encountered in greater
    accept = (lnpdiff > np.log(self._random.rand(len(lnpdiff))))

tests/test_mcmc.py::test_examine_chop_chains
tests/test_priors.py::test_compute_lnprob
  /home/arya/orbitize/orbitize/priors.py:269: RuntimeWarning: invalid value encountered in log
    lnprob = np.log(np.sin(element_array)/normalization)

tests/test_priors.py::test_compute_lnprob
  /home/arya/orbitize/tests/test_priors.py:69: RuntimeWarning: divide by zero encountered in log
    assert np.log(expected_probs[Prior]) == pytest.approx(lnprobs, abs=threshold)

tests/test_priors.py::test_compute_lnprob
  /home/arya/orbitize/orbitize/priors.py:163: RuntimeWarning: divide by zero encountered in log
    lnprob = -np.log((element_array*normalizer))

tests/test_priors.py::test_compute_lnprob
  /home/arya/orbitize/orbitize/priors.py:269: RuntimeWarning: divide by zero encountered in log
    lnprob = np.log(np.sin(element_array)/normalization)

tests/test_priors.py::test_compute_lnprob
  /home/arya/orbitize/orbitize/priors.py:330: RuntimeWarning: divide by zero encountered in log
    lnprob = np.log((self.m*element_array + self.b)/normalizer)

tests/test_priors.py::test_compute_lnprob
  /home/arya/orbitize/orbitize/priors.py:330: RuntimeWarning: invalid value encountered in log
    lnprob = np.log((self.m*element_array + self.b)/normalizer)

tests/test_system.py::test_convert_data_table_radec2seppa
  /home/arya/orbitize/orbitize/system.py:232: RuntimeWarning: invalid value encountered in double_scalars
    pa_err = np.degrees(sep_err/sep)

-- Docs: https://docs.pytest.org/en/latest/warnings.html
============================ 9 failed, 29 passed, 26 warnings in 27.55s ============================
```

**Comment by henry-ngo** [comment]
For the first issue (with np.linspace), we should do a quick bugfix. I checked and this deprecation has been here since numpy v1.12 (latest is v1.17 now, so we should probably get it fixed soon). https://docs.scipy.org/doc/numpy/release.html#id92

**Comment by henry-ngo** [comment]
For the other 9 errors, they seem to all come from assertions on astropy table lengths. Looking at the output, it looks like the length of the table is being computed incorrectly in these tests.

I did a test on my install and did not encounter the same errors---everything passed. Our travis build on the latest commit to master also passed: https://travis-ci.org/sblunt/orbitize/builds/617989705

I am wondering if there was a recent major change in astropy that altered how "length" is computed? @aryadas98 : which astropy version are you running on these tests?

**Comment by henry-ngo** [comment]
(PR #154 addresses the numpy issue)

**Comment by henry-ngo** [comment]
Looks like tests on #154 is now failing, very similar to #151. Something changed with astropy probably in the last 2 months!

**Comment by aryadas98** [comment]
I am using astropy version 4.0 in these tests.

**Comment by henry-ngo** [comment]
It looks like astropy4.0 includes a lot of changes to tables which might make our _compare_tables method not function as expected. However, it also introduces a method to actually compare two tables so that might be a good fix. 

astropy v4 doesn't support python2 so if we change the tests to work with astropy 4 this might mean we have to drop python2 support (alternatively we may have to require an older version of astropy for now). I think we need to think about how we want to address this. See also #132

**Comment by aryadas98** [comment]
Fixed it! The problem was with the `read_file` method in `read_input.py`. Astropy changed how masked tables are dealt with. When a file is read `masked=False` even if the file contains empty cells. https://docs.astropy.org/en/stable/table/masking.html

Adding these two lines just after reading the file fixes the problem:
```
if input_table.has_masked_columns:
    input_table = Table(input_table, masked=True, copy=False)
```

However the property `has_masked_columns` is only available in astropy 4.0

**Comment by aryadas98** [comment]
I could add a check to see the astropy version number first before executing this command. Should I submit a pull request?

**Comment by aryadas98** [comment]
I opened PR #155. The tests are passing.

**Comment by sblunt** [comment]
Hey @aryadas98! Thanks very much for submitting this issue and for finding a fix. The issue you raised sparked a broader discussion about whether to continue supporting Python 2. We've decided to discontinue support for Python 2.7 so that we don't have to have version checks like the one you implemented in PR #155. The plan for the next two versions is:
1. Release a new version that requires astropy < 4 and implements the numpy fix. This will be the last Python 2 compatible version.
2. Immediately afterward, release a new version that requires astropy = 4. This version will not be Python 2 compatible.

**Comment by sblunt** [comment]
Addressed in PR #157
